WordPress/Classes/ViewRelated/Blog/Site Picker/SiteSwitcherView.swift

```diff
@@ -42,51 +42,39 @@ final class SiteSwitcherViewController: UIViewController {
 }
 
 private struct SiteSwitcherView: View {
-    @State private var searchText = ""
     @State private var isSearching = false
 
     let addSiteAction: (() -> Void)
     let onSiteSelected: ((Blog) -> Void)
-    var eventTracker: EventTracker = DefaultEventTracker()
+
+    @StateObject private var viewModel = BlogListViewModel()
 
     var body: some View {
-        if #available(iOS 17.0, *) {
-            blogListView
-                .searchable(
-                    text: $searchText,
-                    isPresented: $isSearching,
-                    placement: .navigationBarDrawer(displayMode: .always)
-                )
-        } else {
-            blogListView
-                .searchable(
-                    text: $searchText,
-                    placement: .navigationBarDrawer(displayMode: .always)
-                )
-                .onChange(of: searchText) { newValue in
-                    isSearching = !newValue.isEmpty
-                }
-        }
+        BlogListView(viewModel: viewModel, onSiteSelected: onSiteSelected)
+            /// - warning: The order is important for `isSearching` to work.
+            .safeAreaInset(edge: .bottom) {
+                SiteSwitcherToolbarView(addSiteAction: addSiteAction)
+            }
+            .searchable(text: $viewModel.searchText, placement: .navigationBarDrawer(displayMode: .always))
+            .navigationTitle(Strings.navigationTitle)
+            .navigationBarTitleDisplayMode(.inline)
     }
+}
 
-    private var blogListView: some View {
-        BlogListView(
-            isSearching: $isSearching,
-            searchText: $searchText,
-            onSiteSelected: onSiteSelected
-        )
-        .safeAreaInset(edge: .bottom) {
-            if !isSearching {
-                HStack {
-                    Spacer()
-                    FAB(action: addSiteAction)
-                        .padding(.trailing, 20)
-                        .padding(.bottom, UIDevice.current.userInterfaceIdiom == .pad ? 20 : 0)
-                }
+private struct SiteSwitcherToolbarView: View {
+    let addSiteAction: (() -> Void)
+
+    @Environment(\.isSearching) var isSearching
```

Comment:
I feel like it's a bit convoluted that this View type uses the `isSearching` env, but with no code that relates to search.

Since the search part exists in its parent view, what do you think changing this `SiteSwitcherToolbarView` type to a function in `SiteSwitcherView`? In that way, the `searchable` modifier and reading `isSearching` env are in the place, which would make the code easier to follow.

Reply:
Unfortunately, it has to go here. If it's in `SiteSwitcherView`, you get the following situation:

```swift
struct SiteSwitcherView: View {
    @Environment(\.isSearching) var isSearching
 
    var body: some View {
        BlogListView(viewModel: viewModel, onSiteSelected: onSiteSelected)
            .safeAreaInset(edge: .bottom) {
                if !isSearching {
                    fab
                }
            }
            // `searchable` gets applied to `BlogListView` and `.safeAreaInset` modifier, but not
            // to the `SiteSwitcherView` itself, so the `isSearching` environment variable will not work
            .searchable(text: $viewModel.searchText)
            // ...
    }
}
```

> type uses the isSearching env, but with no code that relates to search

The environment variables are designed to be used like this, so that's a pretty common approach. You can also see it in the [documentation](https://developer.apple.com/documentation/swiftui/environmentvalues/issearching) that also has a note about the gotcha with where the `isSearching` variable needs to go.

Reply:
Thanks for the link! That's a weird pattern to get it work...